internal/resources/sample_bucket.go

```diff
@@ -0,0 +1,351 @@
+package resources
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api"
+	samplebucket "github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api/sample_bucket"
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/errors"
+	providerschema "github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/schema"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &SampleBucket{}
+	_ resource.ResourceWithConfigure   = &SampleBucket{}
+	_ resource.ResourceWithImportState = &SampleBucket{}
+)
+
+// Samples is the samples resource implementation.
+type SampleBucket struct {
+	*providerschema.Data
+}
+
+// NewSamples is a helper function to simplify the provider implementation.
+func NewSampleBucket() resource.Resource {
+	return &SampleBucket{}
+}
+
+// Metadata returns the samples resource type name.
+func (s *SampleBucket) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_samplebucket"
+}
+
+// Configure It adds the provider configured api to the project resource.
+func (s *SampleBucket) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	data, ok := req.ProviderData.(*providerschema.Data)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *ProviderSourceData, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	s.Data = data
+}
+
+// ImportState imports a remote sample cluster that is not created by Terraform.
+func (s *SampleBucket) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	// Retrieve import ID and save to id attribute
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+// Schema defines the schema for the samples resource.
+func (s *SampleBucket) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = SampleBucketSchema()
+}
+
+func (s *SampleBucket) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan providerschema.Bucket
```

Comment:
Ig we should create a separate struct for the sample Bucket instead of using the same Bucket struct, as both are separate resources.